File: MARN-React-main/src/types/property.ts
export type PropertyStatus = 'available' | 'rented' | 'pending' | 'inactive' | 'Occupied' | 'occupied'

export type PropertyType =
  | 'Apartment'
  | 'House'
  | 'Room'
  | 'SharedRoom'

export type RentalUnit = 'Daily' | 'Monthly' | 'Quarterly' | 'Yearly'

export type SortBy =
  | 'Newest'
  | 'Price'
  | 'Rating'
  | 'Bedrooms'
  | 'Bathrooms'
  | 'SquareMeters'
  | 'Distance'

export const AMENITY_OPTIONS = [
  'Wifi',
  'Parking',
  'AirConditioning',
  'Heating',
  'Tv',
  'Elevator',
  'Pool',
  'Gym',
  'Kitchen',
  'Dishwasher',
  'Microwave',
  'Refrigerator',
  'Washer',
  'Dryer',
  'Balcony',
  'HardwoodFloors',
  'StorageSpace',
  'SecuritySystem',
] as const

export type Amenity = (typeof AMENITY_OPTIONS)[number]

/** Human-readable labels for amenity values */
export const AMENITY_LABELS: Record<Amenity, string> = {
  Wifi: 'WiFi',
  Parking: 'Parking',
  AirConditioning: 'Air Conditioning',
  Heating: 'Heating',
  Tv: 'TV',
  Elevator: 'Elevator',
  Pool: 'Pool',
  Gym: 'Gym',
  Kitchen: 'Kitchen',
  Dishwasher: 'Dishwasher',
  Microwave: 'Microwave',
  Refrigerator: 'Refrigerator',
  Washer: 'Washer',
  Dryer: 'Dryer',
  Balcony: 'Balcony',
  HardwoodFloors: 'Hardwood Floors',
  StorageSpace: 'Storage Space',
  SecuritySystem: 'Security System',
}

export interface Property {
  id: string
  title: string
  description?: string
  type: PropertyType
  typeDisplayName?: string
  status: PropertyStatus
  location: string
  address?: string
  city?: string
  cityDisplayName?: string
  governorate?: string
  governorateDisplayName?: string
  zipCode?: string
  price: number
  beds: number
  bedrooms?: number
  baths: number
  bathrooms?: number
  guests: number
  area?: number
  squareMeters?: number
  images?: string[]
  /** Primary listing image — first entry of images or standalone thumbnail. */
  image?: string
  imagePath?: string
  amenities: any[]
  rules?: any[]
  rating?: number
  averageRating?: number
  reviews?: number
  ratingsCount?: number
  ownerId?: string
  ownerName?: string
  ownerAvatarUrl?: string
  ownerBio?: string
  hostId?: string
  hostName?: string
  hostProfileImage?: string
  hostBio?: string
  hostedBy?: {
    id: string
    fullName: string
    profileImage: string
    averageRating: number
    propertiesCount: number
    bio: string
  }
  propertiesCount?: number
  featured?: boolean
  isShared?: boolean
  rentalUnit?: RentalUnit
  rentalUnitDisplayName?: string
  latitude?: number
  longitude?: number
  createdAt?: string
  updatedAt?: string
  isSaved?: boolean
  maxOccupants?: number
  currentOccupantsCount?: number
  isActive?: boolean
  availability?: boolean
  tenants?: Array<{
    id: string
    fullName: string
    profileImage?: string
  }>
  activeRenters?: Array<{
    id: string
    name: string
    profilePhoto?: string
    matchingPercentage?: number
  }>
  comments?: any[]
  ownerEmail?: string
  hostEmail?: string
}

/** Lightweight shape returned by GET /api/Property/search */
export interface SearchProperty {
  id: number
  imagePath: string
  title: string
  address: string
  latitude?: number
  longitude?: number
  bedrooms: number
  bathrooms: number
  maxOccupants: number
  type: PropertyType
  typeDisplayName?: string
  averageRating: number
  ratings: number
  price: number
  rentalUnit: RentalUnit
  rentalUnitDisplayName?: string
  isSaved: boolean
}

export interface PropertyFilters {
  // text / geo
  keyword?: string
  latitude?: number
  longitude?: number
  radiusKm?: number
  // location enums
  city?: string
  governorate?: string
  // property attributes
  type?: PropertyType
  rentalUnit?: RentalUnit
  isShared?: boolean
  // price
  minPrice?: number
  maxPrice?: number
  // rooms
  minBedrooms?: number
  minBeds?: number
  minBathrooms?: number
  minMaxOccupants?: number
  // area
  minSquareMeters?: number
  maxSquareMeters?: number
  // rating
  minRating?: number
  // amenities (multi-select)
  amenities?: string[]
  // sorting
  sortBy?: SortBy
  sortAscending?: boolean
  // pagination
  page?: number
  pageSize?: number
  // legacy / compat fields kept for other callers
  search?: string
  location?: string
  beds?: number
  baths?: number
  featured?: boolean
}

File: backend MARN/MARN_API/Models/Property.cs
using System;
using System.Collections.Generic;
using MARN_API.Enums.Property;

namespace MARN_API.Models
{
    public class Property
    {
        public long Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public PropertyType Type { get; set; }
        public string ProofOfOwnership { get; set; } = string.Empty;

        public int MaxOccupants { get; set; }
        public bool IsShared { get; set; }

        public int Bedrooms { get; set; }
        public int Beds { get; set; }
        public int Bathrooms { get; set; }
        public double SquareMeters { get; set; }
        public int Views { get; set; } = 0;

        public decimal Price { get; set; }
        public RentalUnit RentalUnit { get; set; }

        public string Address { get; set; } = string.Empty;
        public string City { get; set; } = string.Empty;
        public string State { get; set; } = string.Empty;
        public string ZipCode { get; set; } = string.Empty;
        public double Latitude { get; set; }
        public double Longitude { get; set; }

        public bool IsActive { get; set; } = true;
        public PropertyStatus Status { get; set; } = PropertyStatus.Pending;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? DeletedAt { get; set; }
        public string? ImagesDeletionJob { get; set; }


        public Guid OwnerId { get; set; }
        public virtual ApplicationUser Owner { get; set; } = null!;
        public virtual ICollection<Contract> Contracts { get; set; } = new HashSet<Contract>();
        public virtual ICollection<BookingRequest> BookingRequests { get; set; } = new HashSet<BookingRequest>();
        public virtual ICollection<PropertyFeedback> PropertyFeedbacks { get; set; } = new HashSet<PropertyFeedback>();
        public virtual ICollection<PropertyAmenity> Amenities { get; set; } = new HashSet<PropertyAmenity>();
        public virtual ICollection<PropertyRule> Rules { get; set; } = new HashSet<PropertyRule>();
        public virtual ICollection<PropertyMedia> Media { get; set; } = new HashSet<PropertyMedia>();
        public virtual ICollection<SavedProperty> SavedProperty { get; set; } = new HashSet<SavedProperty>();
    }
}



